app/src/main/java/org/jellyfin/androidtv/ui/startup/ListServerFragment.kt

```diff
@@ -0,0 +1,118 @@
+package org.jellyfin.androidtv.ui.startup
+
+import android.os.Bundle
+import androidx.annotation.DrawableRes
+import androidx.core.view.updatePadding
+import androidx.fragment.app.Fragment
+import androidx.leanback.app.RowsSupportFragment
+import androidx.leanback.widget.ArrayObjectAdapter
+import androidx.leanback.widget.HeaderItem
+import androidx.leanback.widget.ListRow
+import androidx.leanback.widget.ListRowPresenter
+import androidx.leanback.widget.OnItemViewClickedListener
+import androidx.leanback.widget.Presenter
+import androidx.leanback.widget.Row
+import androidx.leanback.widget.RowPresenter
+import androidx.lifecycle.Observer
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.data.model.Server
+import org.jellyfin.androidtv.data.model.ServerList
+import org.jellyfin.androidtv.data.model.User
+import org.jellyfin.androidtv.ui.GridButton
+import org.jellyfin.androidtv.ui.presentation.GridButtonPresenter
+import org.koin.android.viewmodel.ext.android.sharedViewModel
+import timber.log.Timber
+
+class ListServerFragment : RowsSupportFragment() {
+	private companion object {
+		private const val ADD_USER = 1
+		private const val SELECT_USER = 2
+	}
+
+	private val loginViewModel: LoginViewModel by sharedViewModel()
+
+	private val itemViewClickedListener = OnItemViewClickedListener() { _: Presenter.ViewHolder, item: Any, _: RowPresenter.ViewHolder, _: Row ->
+		if (item is UserGridButton) {
+			if (item.user.hasPassword) {
+				// Open login fragment
+				navigate(UserLoginFragment(
+					user = item.user,
+					onConfirmCallback = { username: String, password: String ->
+						GlobalScope.launch {
+							loginViewModel.login(server = item.server, username = username, password = password)
+						}
+					},
+					onCancelCallback = { parentFragmentManager.popBackStack() }
+				))
+			} else {
+				GlobalScope.launch {
+					loginViewModel.login(item.server, item.user.name, "")
+				}
+			}
+		} else if (item is AddUserGridButton) {
+			// Open login fragment
+			navigate(UserLoginFragment(
+				onConfirmCallback = { username: String, password: String ->
+					GlobalScope.launch {
+						loginViewModel.login(server = item.server, username = username, password = password)
+					}
+				},
+				onCancelCallback = { parentFragmentManager.popBackStack() }
+			))
+		}
+	}
+
+	override fun onActivityCreated(savedInstanceState: Bundle?) {
+		super.onActivityCreated(savedInstanceState)
+
+		requireView().updatePadding(top = 20)
```

Comment:
I think this one can be moved to `onCreateView`

Reply:
The viewmodel observer seems to need to be added in `onActivityCreated` otherwise you get some weird crashes when the activity changes.

Reply:
I only meant the padding line